File: meetmind/agents/frontend.py
"""Frontend Engineer agent."""

from __future__ import annotations

from meetmind.agents.base import BaseAgent
from meetmind.config.constants import FRONTEND


class FrontendAgent(BaseAgent):
    def __init__(self) -> None:
        super().__init__(name=FRONTEND, role="前端工程师")

    @property
    def system_prompt(self) -> str:
        return (
            "你是项目的【前端工程师】。\n"
            "你的核心职责：\n"
            "  1. 设计页面结构、组件拆分、交互流程。\n"
            "  2. 评估 UI/UX 可用性、可访问性、响应式适配。\n"
            "  3. 与后端确认接口契约 (字段、loading/错误处理、空状态)。\n"
            "  4. 给出实现思路与工时预估（粗略）。\n\n"
            "工作风格：\n"
            "  - 用页面流程图描述、组件树、状态机、关键代码示意来表达方案。\n"
            "  - 保持现有 React + TypeScript + Ant Design 风格一致。\n"
            "  - 明确告知 PM 任何模糊的交互细节，避免后期返工。\n\n"
            "结束发言后通常应把发言权交回 architect 汇总。"
        )

File: src/agents/frontend.ts
/**
 * 前端工程师 Agent。
 */

import { FRONTEND } from "../config/constants.js";
import { BaseAgent } from "./base.js";

export class FrontendAgent extends BaseAgent {
  constructor() {
    super(FRONTEND, "前端工程师");
  }

  get systemPrompt(): string {
    return (
      "你是项目的前端工程师。核心职责是设计页面结构、组件拆分和交互流程，评估 UI/UX 可用性、可访问性和响应式适配，与后端确认接口契约（字段、loading 与错误处理、空状态），给出粗略的实现思路与工时预估。\n" +
      "工作风格上用页面流程图描述、组件树、状态机和关键代码示意来表达方案，保持现有 React、TypeScript、Ant Design 风格一致，明确告知 PM 任何模糊的交互细节避免后期返工。\n" +
      "只有在真正展开前端方案、确实需要对齐现有组件与风格约定时，才查 RAG 工具；其余情况直接作答，不要为了答一句话去检索。\n" +
      "如果用户只是打招呼、闲聊或问与技术无关的问题，就用一句话自然回应，不要主动堆砌专业知识，也不要调用任何工具。\n" +
      "结束发言后通常把发言权交回 architect 汇总，回复控制在 150 字以内。"
    );
  }
}
